### PR TITLE
Submission

### PR DESCRIPTION
**Pull request by ManickYoj**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1 +1,3 @@
 *.pyc
+*.sublime-project
```
Comment (youralien) [inline .gitignore:2]: aw, why not also gitignore the bag files too?

File: launch/test.launch [hunk 2]
```diff
@@ -0,0 +1,15 @@
+<launch>
```
Comment (youralien) [inline launch/test.launch:1]: good on you for creating a launch file and sending copies to the whole list!  unfortunately, I didn't see a lot of other people benefit by using the launch file.

File: launch/test.launch [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+  <!-- Params/Args -->
+  <arg name="name"/>
+
+  <!-- Run the rosbag -->
+  <node name="rosbag_test" pkg="rosbag" type="play" output="screen" args="-l $(find sign_follower)/bags/$(arg name).bag"/>
+
+  <!-- Decompress bagged camera images -->
+  <node name="republish" type="republish" pkg="image_transport" output="screen" args="compressed in:=/camera/image_raw _image_transport:=compressed raw out:=/camera/image_raw" />
```
Comment (youralien) [inline launch/test.launch:9]: And now I feel really bad for anyone who didn't use a launch file, given that they probably had to start image_transport every single time they were testing!

File: scripts/street_sign_recognizer.py [hunk 4]
```diff
@@ -3,60 +3,102 @@
...
+
+        cv2.namedWindow('raw')
+        cv2.namedWindow('hsv')
+        cv2.namedWindow('bin')
+
+        # Configuration (eg. hsv bounds) are loaded in using a custom config object
+        # The sliders are generated automatically by that config object as well and
+        # any new values are saved automatically at shutdown
+        cfgFilepath = rospy.get_param('~cfgFilepath')
+        self.config = cv2_utils.Config(cfgFilepath)
```
Comment (youralien) [inline scripts/street_sign_recognizer.py:35]: your way of handling the HSV config 'tis beautiful.  I learned a thing.

File: scripts/dict_utils.py [hunk 5]
```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/env python
+
+import functools, operator
+
+# Copied from StackOverflow response
+def getFromDict(data, path):
```
Comment (youralien) [inline scripts/dict_utils.py:6]: These util functions could have definitely used documentation on what an example input might look like.  But I like the functional programming syntax going on in your code, it makes things really succinct.

**Review by youralien — COMMENTED** [review]
I would have argued to use a little more comments, because you do some fancy things and create new helper functions.  

Did the method you end up using work for you?  Did you use cv2.drawMatches in order to debug the issues?

Finally, I am actually really curious to follow up with you about how using ORB and getting the 10 closest matches worked (as opposed to SIFT + Lowe's Ratio test).

**Comment by ManickYoj** [comment]
Ryan - thanks for the detailed feedback. I can up the comments in future to explain my work. Unfortunately, I tend to alternate between next to no comments and commenting every line; never quite settled on that helpful middle ground.

**Pull request closed without merging**